src/physical/columnar/operations/columnscan_equal_value.rs

```diff
@@ -12,21 +22,43 @@ where
     scan: &'a ColumnScanCell<'a, T>,
 
     /// The value the scan jumps to
-    value: T,
+    interval: Interval<T>,
 
-    /// Current value of this scan; can either be `Some(value)` or None
-    current_value: Option<T>,
+    /// Status of this scan.
+    status: ColumnScanStatus,
 }
 impl<'a, T> ColumnScanEqualValue<'a, T>
 where
     T: 'a + ColumnDataType,
 {
     /// Constructs a new [`ColumnScanEqualValue`].
-    pub fn new(scan: &'a ColumnScanCell<'a, T>, value: T) -> ColumnScanEqualValue<'a, T> {
+    pub fn new(
+        scan: &'a ColumnScanCell<'a, T>,
+        interval: Interval<T>,
+    ) -> ColumnScanEqualValue<'a, T> {
         ColumnScanEqualValue {
             scan,
-            value,
-            current_value: None,
+            interval,
+            status: ColumnScanStatus::Before,
+        }
+    }
+
+    fn satisfy_lower_bound(&mut self) -> bool {
```

Comment:
I think it might be helpful in the future to have a comment about the return value, i.e. that the return value indicates that the scan has been moved (at least that's my understanding right now).